warehouse/warehouse.go

```diff
@@ -1125,6 +1129,7 @@ func (wh *HandleT) Setup(whType string) {
 func (wh *HandleT) Shutdown() {
 	wh.backgroundCancel()
 	wh.backgroundWait()
+	jobs.AsyncJobWH.Cancel()
```

Comment:
Let's expose an api in the jobs package like `shutdown or cleanup` or something to cancel the context instead of directly calling it.

Reply:
Addressed this.